In [14]:
from google.colab import files
uploaded = files.upload()


Saving Cleaned_StudentsPerformance.csv to Cleaned_StudentsPerformance (1).csv


In [15]:
import pandas as pd

data = pd.read_csv("Cleaned_StudentsPerformance.csv")   # Agar cleaned file use kar rahi ho
# ya agar original data hai to:
# data = pd.read_csv("StudentsPerformance.csv")

data.head()


gender race/ethnicity parental level of education         lunch  \
0  female        group B           bachelor's degree      standard   
1  female        group C                some college      standard   
2  female        group B             master's degree      standard   
3    male        group A          associate's degree  free/reduced   
4    male        group C                some college      standard   

  test preparation course  math score  reading score  writing score  
0                    none          72             72             74  
1               completed          69             90             88  
2                    none          90             95             93  
3                    none          47             57             44  
4                    none          76             78             75

In [12]:
from statistics import mode

print("Mean Scores:")
print(data[['math score','reading score','writing score']].mean())

print("\nMedian Scores:")
print(data[['math score','reading score','writing score']].median())

print("\nMode Scores:")
print(data[['math score','reading score','writing score']].mode().iloc[0])

print("\nVariance:")
print(data[['math score','reading score','writing score']].var())


Mean Scores:
math score       66.089
reading score    69.169
writing score    68.054
dtype: float64

Median Scores:
math score       66.0
reading score    70.0
writing score    69.0
dtype: float64

Mode Scores:
math score       65
reading score    72
writing score    74
Name: 0, dtype: int64

Variance:
math score       229.918998
reading score    213.165605
writing score    230.907992
dtype: float64


Mean shows average performance

Median shows middle performance

Mode shows most common score

Variance batata hai kitna variation hai scores me

In [11]:
data['average_score'] = (data['math score'] + data['reading score'] + data['writing score']) / 3
data.head()


gender race/ethnicity parental level of education         lunch  \
0  female        group B           bachelor's degree      standard   
1  female        group C                some college      standard   
2  female        group B             master's degree      standard   
3    male        group A          associate's degree  free/reduced   
4    male        group C                some college      standard   

  test preparation course  math score  reading score  writing score  \
0                    none          72             72             74   
1               completed          69             90             88   
2                    none          90             95             93   
3                    none          47             57             44   
4                    none          76             78             75   

   average_score  
0      72.666667  
1      82.333333  
2      92.666667  
3      49.333333  
4      76.333333

In [10]:
numeric_data = data[['math score','reading score','writing score','average_score']]
corr = numeric_data.corr()
corr


math score  reading score  writing score  average_score
math score       1.000000       0.817580       0.802642       0.918746
reading score    0.817580       1.000000       0.954598       0.970331
writing score    0.802642       0.954598       1.000000       0.965667
average_score    0.918746       0.970331       0.965667       1.000000

In [9]:
corr_target = corr['average_score'].sort_values(ascending=False)
corr_target


average_score    1.000000
reading score    0.970331
writing score    0.965667
math score       0.918746
Name: average_score, dtype: float64

The top 3 features most related to student performance are:

Writing Score (Correlation ≈ 0.96)

Reading Score (Correlation ≈ 0.97)

Math Score (Correlation ≈ 0.91)

This means that students who read well also write well and generally perform better academically. Reading and writing abilities are strongly linked to overall performance, while math also contributes but slightly less comparatively.